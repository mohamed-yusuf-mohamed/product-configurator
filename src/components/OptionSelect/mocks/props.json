{
  "selected": {
    "value": "dotted",
    "label": "Dotted",
    "attributes": {
        "cover-colour": {
            "id": "cc1",
            "type": "cover-colour",
            "value": "red",
            "label": "Sunset",
            "selectable": true
        },
        "paper-type": {
            "id": "pt1",
            "type": "paper-type",
            "value": "dotted",
            "label": "Dotted",
            "selectable": true
        }
    }
  },
  "unselected": {
    "value": "dotted",
    "label": "Dotted",
    "attributes": {
        "cover-colour": {
            "id": "cc1",
            "type": "cover-colour",
            "value": "red",
            "label": "Sunset",
            "selectable": true
        },
        "paper-type": {
            "id": "pt1",
            "type": "paper-type",
            "value": "lined",
            "label": "Lined",
            "selectable": true
        }
    }
  }
}