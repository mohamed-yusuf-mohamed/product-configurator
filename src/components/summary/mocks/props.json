{
  "cover-colour": {
      "id": "cc1",
      "type": "cover-colour",
      "value": "red",
      "label": "Sunset",
      "selectable": true
  },
  "paper-type": {
      "id": "pt1",
      "type": "paper-type",
      "value": "dotted",
      "label": "Dotted",
      "selectable": true
  },
  "page-count": {
      "id": "pc1",
      "type": "page-count",
      "value": 170,
      "label": "170",
      "selectable": false
  },
  "price": "Â£17.00",
  "className": "sc-bcXHqe iPrJEm"
}