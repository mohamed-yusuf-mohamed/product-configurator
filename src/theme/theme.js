const theme = {
  title: {
    colour: "hsl(0, 0%, 12%)",
  },
  description: {
    colour: "hsl(0, 0%, 21%)",
  },
  font: "https://fonts.googleapis.com/css2?family=Roboto&display=swap",
  option: {
    button: {
      selected: {
        border: `3px solid hsl(164, 100%, 30%)`,
      },
      unselected: {
        border: "2px solid hsl(0, 0%, 80%)",
      },
      padding: ".7rem",
      border: {
        radius: ".5rem",
      },
    },
  },
  summary: {
    table: {
      border: "1px solid hsl(0, 0%, 80%)",
      cell: {
        padding: "1rem",
      },
    },
  },
}

export default theme
