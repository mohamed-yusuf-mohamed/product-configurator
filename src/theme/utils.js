export const parseHSL = ({ hue, sat, light }) => {
  // const {hue, sat, light} = this.colour.highlight
  return `hsl(${hue}, ${sat}%, ${light}%)`
}
