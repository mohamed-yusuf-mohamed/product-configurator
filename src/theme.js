import { parseHSL } from "./utils";
// Colours are defined in HSL format. https://css-tricks.com/yay-for-hsla

const theme = {
  // configurator: {
  title: {
    colour: "hsl(0, 0%, 12%)",
  },
  description: {
    colour: "hsl(0, 0%, 21%)",
  },
  font: "https://fonts.googleapis.com/css2?family=Roboto&display=swap",
  option: {
    button: {
      selected: {
        border: `1px solid hsl(164, 100%, 30%)`,
      },
      unselected: {
        border: "1px solid hsl(0, 0%, 21%)",
      },
      padding: "20px",
      border: {
        radius: "2rem",
      },
    },
  },
};

export default theme;
