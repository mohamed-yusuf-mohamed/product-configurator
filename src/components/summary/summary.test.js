import theme from "../../theme"
import { ThemeProvider } from "styled-components"
import React from "react"
import { render } from "@testing-library/react"
import Summary from "./index"
import props from "./mocks/props.json"

describe("summary component", () => {
  const { getByTestId } = render(
    <ThemeProvider theme={theme}>
      <Summary {...props} />
    </ThemeProvider>
  )
  const component = getByTestId("product-summary")

  it("renders", () => {
    expect(component).toMatchSnapshot()
  })
})
