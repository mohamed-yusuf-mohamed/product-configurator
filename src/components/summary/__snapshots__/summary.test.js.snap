// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`summary component renders 1`] = `
<div
  class="sc-bcXHqe iPrJEm sc-bcXHqe iPrJEm"
  data-testid="product-summary"
>
  <table>
    <tbody>
      <tr>
        <th>
          Summary
        </th>
        <th
          class="hide"
        >
          Values
        </th>
      </tr>
      <tr>
        <td>
          Pages
        </td>
        <td>
          170
        </td>
      </tr>
      <tr>
        <td>
          Colour
        </td>
        <td>
          Sunset
        </td>
      </tr>
      <tr>
        <td>
          Layout
        </td>
        <td>
          Dotted
        </td>
      </tr>
      <tr>
        <td>
          Price
        </td>
        <td>
          Â£17.00
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
